opflexagent/utils/bridge_managers/vpp_manager.py

```diff
@@ -0,0 +1,124 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from neutron.plugins.ml2.drivers.openvswitch.agent.common import constants
+from opflexagent.utils.bridge_managers import bridge_manager_base
+from opflexagent.utils.bridge_managers import trunk_skeleton
+from oslo_config import cfg
+from oslo_log import log as logging
+from vpplib.VPPApi import VPPApi
+
+
+LOG = logging.getLogger(__name__)
+
+
+class VppManager(bridge_manager_base.BridgeManagerBase,
+                 trunk_skeleton.OpflexTrunkMixin):
+    """ Bridge Manager for VPP."""
+
+    def __init__(self):
+        super(VppManager, self).__init__()
+
+    def initialize(self, host, opflex_conf, agent_state):
+        self.int_br_device_count = 0
+        try:
+            vpp_config = cfg.CONF.VPP
+        except cfg.NoSuchOptError:
+            raise
+        agent_state['vhostuser_socket_dir'] = vpp_config.vhostuser_socket_dir
+        return self, agent_state
+
+    def get_local_ip(self):
+        return None
+
+    def check_bridge_status(self):
+        vapi = VPPApi(LOG, 'gbp-agent')
```

Comment:
I'm not familiar with the VPPApi class -- does it make sense to instantiate it for every call? Should we store the vapi as a member instead?

Reply:
Agree with your comment, this is on my list of things to enhance. As of now we are not monitoring objects in vpp by callbacks, instead relying on querying vpp periodically.  Will need to make the VPP connection a little robust, by checking for the connected status in each call.  I will address this in the following changes.